commit e49b934b77a76443005f92a737dae7370b50e5f7
Author: G. Branden Robinson <g.branden.robinson@gmail.com>
Date:   Tue Jan 23 19:39:03 2024 -0600

    [libgroff]: Fix underspecified getenv() prototype.

    * src/libs/libgroff/getopt.c: Do it.  Seen when building groff on a
      non-glibc-based system (clang 17 complains).

Corepkgs extension for C23:

    * src/libs/libgroff/matherr.c: Define the matherr() parameter in its
      prototype.  C23 no longer supports K&R-style function definitions.

---
Link: https://lists.libreplanet.org/archive/html/groff-commit/2024-01/msg00103.html
 src/libs/libgroff/getopt.c | 2 +-
 src/libs/libgroff/matherr.c | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)
diff --git a/src/libs/libgroff/getopt.c b/src/libs/libgroff/getopt.c
index 6d4ee5b3a..77f8da1ed 100644
--- a/src/libs/libgroff/getopt.c
+++ b/src/libs/libgroff/getopt.c
@@ -124,7 +124,7 @@ static struct _getopt_data getopt_data;
    whose names are inconsistent.  */
 
 #ifndef getenv
-extern char *getenv ();
+extern char *getenv (const char *);
 #endif
 
 #endif /* not __GNU_LIBRARY__ */
diff --git a/src/libs/libgroff/matherr.c b/src/libs/libgroff/matherr.c
--- a/src/libs/libgroff/matherr.c
+++ b/src/libs/libgroff/matherr.c
@@ -29,3 +29,2 @@
-int matherr(exc)
-struct exception *exc;
+int matherr(struct exception *exc)
 {
   switch (exc->type) {
   case SING:
